lib/ipmi_fru.c

```diff
@@ -4797,6 +4797,32 @@ f_type, uint8_t f_index, char *f_string)
 	/* Convert index from character to decimal */
 	f_index= f_index - 0x30;
 
+	if (f_type == 'b' && f_index == 9)		// use index '9' for editing 'board mfg date'
+	{
+		time_t total_secs;
+		uint32_t *board_mfg_date=&fru_data[3];
+
+		total_secs = ipmi_strptime("%Y-%m-%d %H:%M:%S", f_string);
+		ipmi_time2fru(total_secs, board_mfg_date);
+
+		checksum = 0;
+	/* Calculate Header Checksum */
+		for (i = 0; i < fru_section_len - 1; i++)
+		{
+			checksum += fru_data[i];
+		}
+		checksum = (~checksum) + 1;
+		fru_data[fru_section_len - 1] = checksum;
```

Comment:
use `ipmi_csum()` please

Reply:
Also, it looks to me like all this FRU update code forgets to update the overall FRU checksum after modifying an area.

Reply:
since we update the mfg date only and we do not change any length of the records. 
I don't think the common header field checksum will be changed.